keps/sig-node/753-sidecar-containers/README.md

```diff
@@ -843,24 +843,53 @@ examples:
 
 ### Termination of containers
 
-In alpha sidecar containers will be terminated as regular containers. No special
+In Alpha sidecar containers will be terminated as regular containers. No special
 or additional signals will be supported.
 
-We will collect feedback from alpha implementation and plan to improve
-termination in Beta. When Pods with sidecars are terminated:
+In Beta we have thought about introducing additional termination grace period fields
+to manage termination duration
+([draft proposal](https://docs.google.com/document/d/1B01EdgWJAfkT3l6CIwNwoskQ71eyuwet3mjiQrMQbU8))
+and leverage these fields to add reverse order termination of sidecar containers
+after the primary containers terminate.
 
+However, we decided on an alternative that doesn't require additional fields or hooks while keeping
+the desired behaviors when Pods with sidecars are terminated. While original approach works better
+with truly graceful termination where consistency is more important than time taken, proposed approach
```

Comment:
this sentence may need some work to make it easier-readable.

Reply:
+1

More detailed rationale would also be worthwhile here.

The trade off between guaranteeing a per-sidecar grace period and guaranteeing a pod level grade period is one of the biggest implications to this PR (the other being the sidecar restart  behavior after main container shutdown).

Specifically, I think we should acknowledge here in the KEP that some sidecars benefit from being able to flush data to external systems during shutdown. Logging and monitoring sidecars are prime examples.  These sidecars are impacted by the decision this PR codifies.

Could we either add an entry to the alternatives considered that summarizes the idea of providing per-sidecar graceful termination periods and explains the pros, cons and the rationale for why we're going this way instead? 

It might also be worthwhile to add some use cases, both for the usage patterns that benefit from this decision and those that do not. That  would be a good place to describe how the use cases that don't benefit should cope with the design.  What would be the best practices for a logging sidecar given this design?